ruma-serde/src/canonical_json.rs

```diff
@@ -45,12 +47,27 @@ impl fmt::Display for Error {
 
 impl std::error::Error for Error {}
 
+/// Fallible conversion from a `serde_json::Map` to a `CanonicalJsonObject`.
+pub fn try_from_json_map(
+    json: JsonObject<String, JsonValue>,
+) -> Result<CanonicalJsonObject, Error> {
+    json.into_iter().map(|(k, v)| Ok((k, v.try_into()?))).collect()
+}
+
+/// Fallible conversion from any value that impl's `Serialize` to a `CanonicalJsonValue`.
+pub fn to_canonical_value<T: Serialize>(value: T) -> Result<value::CanonicalJsonValue, Error> {
```

Comment:
Having read https://matklad.github.io/2020/10/15/study-of-std-io-error.html recently, I am very skeptical of usage of one error type for JsonValue -> CanonicalJsonValue conversion as well as serde stuff. But that's a pre-existing issue so I'll open a separate issue.